Modules/_functoolsmodule.c

```diff
@@ -932,15 +932,32 @@ _functools_cmp_to_key_impl(PyObject *module, PyObject *mycmp)
 
 /* reduce (used to be a builtin) ********************************************/
 
-// Not converted to argument clinic, because of `args` in-place modification.
-// AC will affect performance.
+/*[clinic input]
+@text_signature "($module, function, iterable[, initial], /)"
```

Comment:
With Serhyi's suggestion we don't actually need `@text_signature` any more. If you remove it, you get:

```
>>> functools.reduce.__text_signature__
'($module, function, iterable, initial=<unrepresentable>, /)'
```

There is an open issue #87233 about supporting such signatures in inspect. I suggest we remove this line for now so that the signature becomes representable once that issue is fixed.
```suggestion
```

Reply:
I'm -0 on this - lets reviewers decide if this suggestion should be applied.

To me, using ``@text_signature`` basically means: FIXME, we actually can't represent signature for this function.  I think that in most (all?) cases fixing this will require support for multiple signatures.  So, the clinic input must be manually modified.  And ``@text_signature`` marks are easily greppable.

Reply:
I'm fine with both Serhiy's suggestion and `@text_signature`. Since both Jelle and Alex is in favour of Serhiy's suggestion, let's go for that.

Reply:
Alex?

Reply:
> Alex?

Yes, he reacted with a thumbs-up to Jelle's https://github.com/python/cpython/pull/125999#discussion_r1817966240